qiskit/algorithms/evolvers/pvqd.py

```diff
@@ -0,0 +1,309 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The projected Variational Quantum Dynamics Algorithm."""
+
+from typing import Optional, Union, List, Tuple, Callable
+
+import numpy as np
+
+from qiskit import transpile
+from qiskit.exceptions import QiskitError
+from qiskit.algorithms.optimizers import Optimizer
+from qiskit.circuit import QuantumCircuit, ParameterVector, ParameterExpression, Parameter
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend
+from qiskit.opflow import (
+    OperatorBase,
+    CircuitSampler,
+    ExpectationBase,
+    ListOp,
+    StateFn,
+)
+from qiskit.opflow.gradients.circuit_gradients import ParamShift
+from qiskit.synthesis import EvolutionSynthesis, LieTrotter
+from qiskit.utils import QuantumInstance
+
+from .evolution_problem import EvolutionProblem
+from .evolution_result import EvolutionResult
+from .real_evolver import RealEvolver
+
+
+class PVQDResult(EvolutionResult):
+    """The result object for the pVQD algorithm."""
+
+    times = None
+    parameters = None
+    fidelities = None
+    estimated_error = None
+    observables = None
+
+
+class PVQD(RealEvolver):
```

Comment:
On a second thought, maybe PVQD is the right name after all because that's what it's known as and what people will look for. And as @woodsp-ibm pointed out -- they do live in different submodules and the docstrings and types are pretty clear on what's what 😄 If we rename it, how would people know what to look for? 🤔

Reply:
Ok, perhaps we could change the deflation acronym then.

Reply:
From a quick Google scholar and paper search "VQD" is also what the deflation algorithm is known as in literature. The "VQ" term in quantum computing is overloaded, users already have to know what they want between the VQE (eigensolver), VQC (ML classifier) and VQD. Since PVQD and VQD also don't take the same inputs I feel like it should be safe to have both, if these are the names they are well-known as.

Reply:
Ok, makes sense.